### ISSUE TITLE
Rating calculation fails

### ISSUE THREAD

**Issue by tomaszmrugalski** [gut-space/svarog#72]
Rating calculation now consistently fails on both TKIS-1 and SKORCZ-1 stations. Here's the exception (this one is from TKIS-1, but I think the same thing happens on SKORCZ-1):

```
2022-01-09 11:26:05,672 INFO receiver.py:52: Starting receiver job: name=NOAA 18 los=2022-01-09T10:39:06.684699, PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/pi/.local/bin
2022-01-09 11:26:05,690 INFO factory.py:118: Recipe will store all files in /home/pi/observations/2022-01-09-1039-noaa-18
2022-01-09 11:26:05,911 INFO sh.py:670: <Command '/usr/bin/rtl_fm -f 137.9125e6 -s 48000 -g 48 -p 1 -F 9 -T -A fast -E DC /home/pi/observations/2022-01-09-1039-noaa-18/signal.raw', pid 30114>: process started
2022-01-09 11:39:08,038 INFO sh.py:670: <Command '/usr/bin/sox -t raw -b16 -es -r 48000 -c1 -V3 /home/pi/observations/2022-01-09-1039-noaa-18/signal.raw /home/pi/observations/2022-01-09-1039-noaa-18/signal.wav rate 11025', pid 30151>: process started
2022-01-09 11:39:10,781 INFO sh.py:670: <Command '/home/pi/.local/bin/noaa-apt -o /home/pi/observations/2022-01-09-1039-noaa-18/product.png /home/pi/observations/2022-01-09-1039-noaa-18/signal.wav', pid 30156>: process started
2022-01-09 11:39:24,536 INFO receiver.py:76: Recipe execution complete, generated 4 result[s] (SIGNAL:/home/pi/observations/2022-01-09-1039-noaa-18/signal.wav PRODUCT:/home/pi/observations/2022-01-09-1039-noaa-18/product.png LOG:/home/pi/observations/2022-01-09-1039-noaa-18/session.log RAW:/home/pi/observations/2022-01-09-1039-noaa-18/signal.raw ), stored in /home/pi/observations/2022-01-09-1039-noaa-18 directory.
2022-01-09 11:39:24,536 INFO receiver.py:91: Submitting results
2022-01-09 11:39:24,536 INFO receiver.py:94: Getting rating for product /home/pi/observations/2022-01-09-1039-noaa-18/product.png (rating algorithm is analog)
2022-01-09 11:39:25,212 ERROR receiver.py:45: Error during rating the product
Traceback (most recent call last):
  File "/home/pi/devel/svarog/station/receiver.py", line 43, in get_rating_for_product
    return rate(img)
  File "/home/pi/devel/svarog/station/quality_ratings/analog.py", line 14, in rate
    return 1.0 - estimate_analog_noise(img)
  File "/home/pi/devel/svarog/station/src/analog-noise-estimator/analog_noise_estimator/main.py", line 15, in estimate
    estimation = estimate_in_boxes(I, L3, box=box)
  File "/home/pi/devel/svarog/station/src/analog-noise-estimator/analog_noise_estimator/estimation.py", line 66, in estimate_in_boxes
    rows, cols = I.shape
ValueError: too many values to unpack (expected 2)
2022-01-09 11:39:25,226 INFO receiver.py:96: Product /home/pi/observations/2022-01-09-1039-noaa-18/product.png got rating None
2022-01-09 11:39:25,291 INFO submitobs.py:151: Submitting observation, url=https://svarog.space/receive, file(s)=['/home/pi/observations/2022-01-09-1039-noaa-18/product.png']
2022-01-09 11:39:27,111 INFO submitobs.py:160: Response status: 204
2022-01-09 11:39:27,113 INFO submitobs.py:171: Upload successful.
```

**Comment by tomaszmrugalski** [comment]
This started happening after TKiS-1 was reinstalled. It never worked on SKORCZ-1. I think this may have to do with migration to python 3.9.

**Comment by tomaszmrugalski** [comment]
This may do the trick. Testing now on TKIS-1 if it will work.

```diff
diff --git a/analog_noise_estimator/estimation.py b/analog_noise_estimator/estimation.py
index 3ee54ab..aebb557 100644
--- a/analog_noise_estimator/estimation.py
+++ b/analog_noise_estimator/estimation.py
@@ -63,7 +63,12 @@ def estimate_in_boxes(I, L, conv2d=np_fftconvolve, box=25):
     pixel of box and 'noise' with estimated noise sigma value (sigma from Gauss
     distribution of noise).
     '''
-    rows, cols = I.shape
+
+    # This is a fix for handling images read by imread from matplotlib.pyplot. Its shape
+    # returns a tuple of 3 elements, instead of two as opencv's imread does. We don't care
+    # about the third parameter, but we don't want python to freak out that we unpack too few
+    # variables.
+    rows, cols, *_ = I.shape
     results = []
     for rowIdx in range(0, rows, box):
         for colIdx in range(0, cols, box):
```

**Comment by tomaszmrugalski** [comment]
The problem is more complex. cv2's and pyplot's imread return a very different data. cv2 returns 2D array of int values 0..255 (image is ready as grayscale), while pyplot's imread returns 3D array, so each pixel is represented as 4 values, which look like RGBA. Value of each channel is a float changing in the 0.0.... 1.0 range.

Here's a bit of code to demonstrate it:
```python
import cv2
from matplotlib import pyplot

img1 = cv2.imread("product.png", cv2.IMREAD_GRAYSCALE)
img2 = pyplot.imread("product.png")

print(f"cv2: shape: {img1.shape}, type of data: {type(img1[0,0])}, first 20 pixels or row 0: {img1[0, 0:19]}")
print(f"pyplot: shape: {img2.shape}, type of data: {type(img2[0,0])}, first 20 pixels or row 0: {img2[0, 0:19]}")
```

It returns this:

```
$ python img_load_test.py 
cv2: shape: (1070, 2080), type of data: <class 'numpy.uint8'>, first 20 pixels or row 0: [  0   0   0  63  51  42 148   1 166 228  68  16 142 149  31  48 113 110
  75]
pyplot: shape: (1070, 2080, 4), type of data: <class 'numpy.ndarray'>, first 20 pixels or row 0: [[0.         0.         0.         1.        ]
 [0.         0.         0.         1.        ]
 [0.         0.         0.         1.        ]
 [0.24705882 0.24705882 0.24705882 1.        ]
 [0.2        0.2        0.2        1.        ]
 [0.16470589 0.16470589 0.16470589 1.        ]
 [0.5803922  0.5803922  0.5803922  1.        ]
 [0.00392157 0.00392157 0.00392157 1.        ]
 [0.6509804  0.6509804  0.6509804  1.        ]
 [0.89411765 0.89411765 0.89411765 1.        ]
 [0.26666668 0.26666668 0.26666668 1.        ]
 [0.0627451  0.0627451  0.0627451  1.        ]
 [0.5568628  0.5568628  0.5568628  1.        ]
 [0.58431375 0.58431375 0.58431375 1.        ]
 [0.12156863 0.12156863 0.12156863 1.        ]
 [0.1882353  0.1882353  0.1882353  1.        ]
 [0.44313726 0.44313726 0.44313726 1.        ]
 [0.43137255 0.43137255 0.43137255 1.        ]
 [0.29411766 0.29411766 0.29411766 1.        ]]
```

**Comment by tomaszmrugalski** [comment]
Testing a solution on TKIS-1... will report results soon.

**Comment by tomaszmrugalski** [comment]
Fix merged in #76. The rating is now being calculated properly, at least on TKIS-1.
